### PR TITLE
Add docs for zpages

### PR DESCRIPTION
**Pull request by richabanker**
### Description

Docs for Kubernetes z-pages 

### Issue

https://github.com/kubernetes/enhancements/issues/4827
https://github.com/kubernetes/enhancements/issues/4828

### CODE REVIEW

**Review by dipesh-rawat — COMMENTED** [review]
@richabanker We should consider setting the **_dev-1.32_** branch as the target branch rather than the _main_ branch for this pull request. Since the change is related to the feature targeting v1.32 release.

**Comment by richabanker** [comment]
> @richabanker We should consider setting the **_dev-1.32_** branch as the target branch rather than the _main_ branch for this pull request. Since the change is related to the feature targeting v1.32 release.

Done

**Comment by rdalbuquerque** [comment]
Hello @richabanker and @kubernetes/sig-docs-en-owners  :wave:! I'm reaching out from the Docs team. Just checking in as we approach Docs Freeze on Tuesday November 26th 18:00 PDT. This documentation appears to still be under review. To meet the Docs Freeze, this PR must have a technical review as well as lgtm and approve labels applied, without any unaddressed comments or concerns from SIG Docs. Thank you!

File: content/en/docs/reference/command-line-tools-reference/feature-gates/component-flagz.md
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/component-flagz.md:1]: This shouldn't be empty.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/component-statusz.md
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/component-statusz.md:1]: This shouldn't be empty.

File: content/en/docs/reference/instrumentation/zpages.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+{{< feature-state for_k8s_version="v1.32" state="alpha" >}}
+
+Kubernetes core components expose a suite of z-endpoints to make it easier for users to debug their components. As of 1.32, these endpoints only support plain text format and are strictly to be used for human inspection to gain real time debugging information of a component binary. Avoid automated scraping of data returned by these endpoints since they are introduced as alpha features and their format may change in future releases.  
+
+<!-- body -->
+
+## z-pages
+
+Kubernetes now includes "z-pages" to help you troubleshoot problems with its core control plane components. These special debugging endpoints provide internal information about running components. Currently we have the following endpoints:
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/zpages.md:19]:
```suggestion
Kubernetes v{{< skew currentVersion >}} allows you to enable _z-pages_ to help you troubleshoot
problems with its core control plane components. These special debugging endpoints provide internal
information about running components. For Kubernetes {{< skew currentVersion >}}, components
serve the following endpoints (when enabled):
```

File: content/en/docs/reference/instrumentation/zpages.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+- [z-pages](#z-pages)
+  - [statusz](#statusz)
+  - [flagz](#flagz)
+
+### statusz
+
+Enabled using the ComponentStatusz feature gate, statusz endpoint displays the high level information about the component such as its Kubernetes version, emulation version, start time and more.
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/zpages.md:27]:
```suggestion
Enabled using the `ComponentStatusz` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/),
the `/statusz` endpoint displays high level information about the component such as its Kubernetes version, emulation version, start time and more.
```

File: content/en/docs/reference/instrumentation/zpages.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+Started: Wed Oct 16 21:03:43 UTC 2024
+Up: 0 hr 00 min 16 sec
+Go version: go1.23.2
+Binary version: 1.32.0-alpha.0.1484&#43;5eeac4f21a491b-dirty
+Emulation version: 1.32.0-alpha.0.1484
+```
+
+### flagz
+
+Enabled using the ComponentFlagz feature gate, the flagz endpoint shows you the command line flags that were used to start a component.
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/zpages.md:44]:
```suggestion
Enabled using the `ComponentFlagz` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/), the `/flagz` endpoint shows you the command line arguments that were used to start a component.
```

File: content/en/docs/reference/instrumentation/zpages.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+The flagz data for apiserver looks like this:
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/zpages.md:46]:
```suggestion
The `/flagz` data for the API server looks something like:
```

File: content/en/docs/reference/instrumentation/zpages.md (hunk 3 above)
Comment (sftim) [reply content/en/docs/reference/instrumentation/zpages.md:44]: Although Go calls all command line arguments "flags", this isn't true across the world of IT. In documentation (not just K8s) I consistently call for the more widely recognized term. We can still talk about flagz when talking about the endpoint.

File: content/en/docs/reference/instrumentation/zpages.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+- [z-pages](#z-pages)
+  - [statusz](#statusz)
+  - [flagz](#flagz)
+
+### statusz
+
+Enabled using the ComponentStatusz feature gate, statusz endpoint displays the high level information about the component such as its Kubernetes version, emulation version, start time and more.
+
+The statusz data for apiserver looks like this:
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/zpages.md:29]:
```suggestion
The `/statusz` response from the API server is similar to:
```

File: content/en/docs/reference/instrumentation/zpages.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+title: Kubernetes z-pages
+content_type: reference
+weight: 20
+---
+
+<!-- overview -->
+
+{{< feature-state for_k8s_version="v1.32" state="alpha" >}}
+
+Kubernetes core components expose a suite of z-endpoints to make it easier for users to debug their components. As of 1.32, these endpoints only support plain text format and are strictly to be used for human inspection to gain real time debugging information of a component binary. Avoid automated scraping of data returned by these endpoints since they are introduced as alpha features and their format may change in future releases.
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/zpages.md:13]:
```suggestion
Kubernetes core components can expose a suite of _z-endpoints_ to make it easier for users
to debug their cluster and its components. These endpoints are strictly to be used for human
inspection to gain real time debugging information of a component binary.
Avoid automated scraping of data returned by these endpoints; in Kubernetes {{< skew currentVersion >}}
these are an **alpha** feature and the response format may change in future releases.
```

File: content/en/docs/reference/instrumentation/zpages.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+<!-- body -->
+
+## z-pages
+
+Kubernetes now includes "z-pages" to help you troubleshoot problems with its core control plane components. These special debugging endpoints provide internal information about running components. Currently we have the following endpoints:
+
+- [z-pages](#z-pages)
+  - [statusz](#statusz)
+  - [flagz](#flagz)
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/zpages.md:23]: Is this right:
```suggestion
  - [`/statusz`](#statusz)
  - [`/flagz`](#flagz)
```
?

File: content/en/docs/reference/instrumentation/zpages.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
+---
+reviewers:
+- richabanker
+title: Kubernetes z-pages
+content_type: reference
+weight: 20
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/zpages.md:6]: Is this right? I'd put it last in the section.

**Review by sftim — CHANGES_REQUESTED** [review]
Thanks, but: the feature gate definitions are missing.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/component-flagz.md
Comment (richabanker) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/component-flagz.md:1]: fixed now, thanks

File: content/en/docs/reference/command-line-tools-reference/feature-gates/component-statusz.md
Comment (richabanker) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/component-statusz.md:1]: fixed

File: content/en/docs/reference/instrumentation/zpages.md (hunk 8 above)
Comment (richabanker) [reply content/en/docs/reference/instrumentation/zpages.md]: Moved to the last section. @dashpole added you as the reviewer, could you take a look please?

File: content/en/docs/reference/instrumentation/zpages.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+default-watch-cache-size=100
+```
+
+---
+reviewers:
+- dashpole
+title: Kubernetes z-pages
+content_type: reference
+weight: 20
+---
```
Comment (sftim) [inline content/en/docs/reference/instrumentation/zpages.md:68]: The [front matter](https://gohugo.io/content-management/front-matter/) needs to come first in the file; that's what Hugo expects.

**Review by sftim — CHANGES_REQUESTED** [review]

File: content/en/docs/reference/instrumentation/zpages.md (hunk 9 above)
Comment (richabanker) [reply content/en/docs/reference/instrumentation/zpages.md]: Ah clearly misunderstood your [previous comment](https://github.com/kubernetes/website/pull/48760#discussion_r1855648814). Tried to fix now, is this better?

File: content/en/docs/reference/instrumentation/zpages.md (hunk 8 above)
Comment (sftim) [reply content/en/docs/reference/instrumentation/zpages.md:4]:
```suggestion
weight: 60
```
This will make it appear last in the list.

**Comment by sftim** [comment]
BTW, for beta we should definitely document what access control looks like for these endpoints. Even if we think it's obvious, the people reading the docs might not agree.

**Review by dashpole — APPROVED** [review]
lgtm for sig-instrumentation

**Pull request merged**